pkg/npm/bom.go

```diff
@@ -0,0 +1,175 @@
+package npm
+
+import (
+	"fmt"
+	"path/filepath"
+	"runtime"
+
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+)
+
+const (
+	// Output file names
+	npmBomFilename  = "bom-npm.xml"
+	tempBomFilename = "bom-npm.json"
+
+	// Package versions
+	cycloneDxNpmPackageVersion = "@cyclonedx/cyclonedx-npm@1.11.0"
+	cycloneDxBomPackageVersion = "@cyclonedx/bom@^3.10.6"
+	cdxgenPackageVersion       = "@cyclonedx/cdxgen@^11.3.2"
+	cycloneDxCliVersion        = "v0.27.2"
+
+	// Configuration
+	cycloneDxNpmInstallationFolder = "./tmp" // This folder is also added to npmignore in publish.go
+	cycloneDxSchemaVersion         = "1.4"
+)
+
+var cycloneDxCliUrl = map[struct{ os, arch string }]string{
+	{"windows", "amd64"}: "https://github.com/CycloneDX/cyclonedx-cli/releases/download/%s/cyclonedx-win-x64.exe",
+	{"windows", "arm64"}: "https://github.com/CycloneDX/cyclonedx-cli/releases/download/%s/cyclonedx-win-arm64.exe",
```

Comment:
I think we don't support running Piper on a windows machine

Reply:
it shouldn't hurt, but if someone looks into the source code he might think that windows is supported by Piper